### PR TITLE
Wishlist and Cart Pages

### PR DESCRIPTION
**Pull request by samyukthasudhakar**
This PR contains wishlist and cart pages with functionality.

The following files need to be reviewed :

#### WishList
- [src/pages/WishList/wishList.jsx](https://github.com/samyukthasudhakar/paws-and-beyond-React/pull/4/files#diff-22cf03b114c7d3dddd639ba1dfa6f579730186749ecf5f3e9e9d6866f70455fa)
- [src/reducers/wishlistReducer.jsx](https://github.com/samyukthasudhakar/paws-and-beyond-React/pull/4/files#diff-28561d45d7bfdaaba3771fc2d0672ce73bb2c6d06646be9d7e50f39eb4ff941f)
- [src/context/wishListContext.jsx](https://github.com/samyukthasudhakar/paws-and-beyond-React/pull/4/files#diff-313220e0b85d47af916c7ea3475ee519534c0cba81bc4e0312ea453f892d9fdc)

#### Cart
- [src/pages/Cart/cart.jsx](https://github.com/samyukthasudhakar/paws-and-beyond-React/pull/4/files#diff-d7a93164fefeee395dffcf1bb65177bc1329f8eae72dc81554929f48e01bdfbe)
- [src/reducers/cartReducer.jsx](https://github.com/samyukthasudhakar/paws-and-beyond-React/pull/4/files#diff-b9316ed261e2a89718f5b110b6e20614d3908151ea14db0fc6c12f054a3c2bf9)
- [src/context/cartContext.jsx](https://github.com/samyukthasudhakar/paws-and-beyond-React/pull/4/files#diff-3c7fe3726ad3af481e202923ebdb69e19efc60b58e2989db6fcf8f52f8caa2f8)

#### Components
- [src/components/card/cartCard/cartCard.jsx](https://github.com/samyukthasudhakar/paws-and-beyond-React/pull/4/files#diff-a81afeb3e59a9fdcbec0e4076c5423e00dffaa3307a34ae0893e4a8102cbb68f)

Live Link : https://cart-and-wishlist-ecommerce.netlify.app/

### CODE REVIEW

File: src/pages/Cart/cart.jsx [hunk 1]
```diff
@@ -1,11 +1,76 @@
...
+import './cart.css'
+import { Header, Footer, CartCard, EmptySection } from '../../components'
+import { emptyBag } from '../../assets/images';
+import { useCart } from '../../context'
 
 function CartPage(){
+    
+    const { cartItems, setCartItems } = useCart()
+    let cartProductsCount = cartItems.length
+    let totalAmount = cartItems ? cartItems.reduce((acc, curr) => acc + curr.price * curr.cartCount, 0) : 0;
```
Comment (foolhardy21) [inline src/pages/Cart/cart.jsx:12]: a minor suggestion - you may abstract these methods in a util file.

File: src/context/wishListContext.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { createContext, useContext, useReducer } from 'react';
+import { WishListReducer } from '../../src/reducers/wishlistReducer';
+
+const wishListContext = createContext(null);
+
+function WishListProvider( { children } ){
+
+    const [ wishList, setWishList ] = useReducer( WishListReducer, [])
```
Comment (foolhardy21) [inline src/context/wishListContext.jsx:8]: suggestion - here `setWishlist` is giving an idea that it's a `useState` but for `useReducer`, as a convention, it is named as `dispatch` or `somethingDispatch` .

**Review by foolhardy21 — APPROVED** [review]
the functionalities are working good. nice work. gave some suggestions that you can include according to your workflow.

File: src/context/cartContext.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+const cartContext = createContext(null);
+
+function CartProvider( { children } ){
+
+    const [ cartItems, setCartItems ] = useReducer(CartReducer, [])
+
+    return (
+        <cartContext.Provider value={{ cartItems, setCartItems}}>
+            { children }
```
Comment (anik31) [inline src/context/cartContext.jsx:12]: appreciation - liked that you're using separate context for wishlist & cart

File: src/reducers/cartReducer.jsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            return [...state, {...action.payload, addedToCart:true, cartCount:1}]
+        case 'REMOVE_FROM_CART':
+            return [...state.filter(product => product._id!==action.payload._id)]
+        case 'INCREMENT_PRODUCT_COUNT':
+            return[ ...state.map(product => (product._id===action.payload._id)?{...product, cartCount: product.cartCount + 1}:product)]
+        case 'DECREMENT_PRODUCT_COUNT':{
+            if (action.payload.cartCount===1){
+                return [...state.filter(product => product._id!==action.payload._id)]
+            }else{
+                return[ ...state.map(product => (product._id===action.payload._id)?{...product, cartCount: product.cartCount - 1}:product)]
```
Comment (anik31) [inline src/reducers/cartReducer.jsx:15]: suggestion - these operations are being handled by mockBee APIs, so we don't need to write the logic separately.

**Review by anik31 — COMMENTED** [review]
Nice work, code is well structured & readable.

suggestion - in cart page you haven't added "remove from cart" for each item, which is a requirement in the prd

**Comment by samyukthasudhakar** [comment]
> the functionalities are working good. nice work. gave some suggestions that you can include according to your workflow.

Thank you, will work on the suggestions :D

**Comment by samyukthasudhakar** [comment]
> Nice work, code is well structured & readable.
> 
Thank you !

> suggestion - in cart page you haven't added "remove from cart" for each item, which is a requirement in the prd

I've added remove cart from functionality to the dismiss button, on the top corner of the cart and as well as when the count goes less the zero it gets auto removed from cart.

File: src/reducers/cartReducer.jsx (hunk 4 above)
Comment (samyukthasudhakar) [reply src/reducers/cartReducer.jsx:15]: Sure, will change them up while integrating with the API's.

File: src/pages/Cart/cart.jsx (hunk 1 above)
Comment (samyukthasudhakar) [reply src/pages/Cart/cart.jsx:12]: sure, will do !

**Pull request merged**